In [75]:
%pip install tensorflow

from tensorflow.keras.preprocessing.text import one_hot

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


# Example usage of one_hot encoding

In [76]:


sent = [
    'the glass of milk',
    'the glass of juice',
    'the cup of tea',
    'I am a good boy',
    'I am a good developer',
    'understand the meaning of words',
    'your videos are good'
    
]

In [77]:
sent

['the glass of milk',
 'the glass of juice',
 'the cup of tea',
 'I am a good boy',
 'I am a good developer',
 'understand the meaning of words',
 'your videos are good']

# Define the  ------------ vocabulary size

In [78]:
voc_size = 10000

# one-hot Representation

In [79]:
sent

['the glass of milk',
 'the glass of juice',
 'the cup of tea',
 'I am a good boy',
 'I am a good developer',
 'understand the meaning of words',
 'your videos are good']

In [80]:
one_hot_repr = [one_hot(sentence, voc_size) for sentence in sent]

one_hot_repr

[[9118, 2198, 319, 5989],
 [9118, 2198, 319, 2891],
 [9118, 7041, 319, 1082],
 [5184, 7907, 6344, 4626, 115],
 [5184, 7907, 6344, 4626, 8886],
 [9827, 9118, 5749, 319, 1358],
 [7968, 6120, 771, 4626]]

#  the glass of milk  is -- coverted into this 10000 vectors == 

[9118, 2198, 319, 5989]

# word embedding representation

In [81]:
from tensorflow.keras.layers import Embedding
# from tensorflow.keras.preprocessing.sequence import pad_sequences
from tensorflow.keras.models import Sequential
from tensorflow.keras.utils import pad_sequences

In [82]:
import numpy as np

# because all words are in diff sizes like 5 , 4 etc to resize same length

sent_length = 8

if any sentence is 5 then 3 zero will be added to make it 8 length add 'pre' padding for same size 

In [83]:
sent_length = 8
embedding_docs = pad_sequences(one_hot_repr, padding='pre', maxlen=sent_length)

print(embedding_docs)

[[   0    0    0    0 9118 2198  319 5989]
 [   0    0    0    0 9118 2198  319 2891]
 [   0    0    0    0 9118 7041  319 1082]
 [   0    0    0 5184 7907 6344 4626  115]
 [   0    0    0 5184 7907 6344 4626 8886]
 [   0    0    0 9827 9118 5749  319 1358]
 [   0    0    0    0 7968 6120  771 4626]]


In [84]:
# sent_length = 8
# embedding_docs = pad_sequences(one_hot_repr, padding='post', maxlen=sent_length)

# print(embedding_docs)

# feature representation

# dimension

In [85]:
dim = 10

In [87]:
model = Sequential()

model.add(Embedding(voc_size, dim, input_length=sent_length))

model.compile(optimizer='adam', loss='mse')



In [ ]:
model

<Sequential name=sequential_7, built=True>

In [ ]:
model.build(input_shape=(None, sent_length))
model.summary()

Model: "sequential_7"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Layer (type)                    ┃ Output Shape           ┃       Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ embedding_7 (Embedding)         │ (None, 8, 10)          │       100,000 │
└─────────────────────────────────┴────────────────────────┴───────────────┘

 Total params: 100,000 (390.62 KB)

 Trainable params: 100,000 (390.62 KB)

 Non-trainable params: 0 (0.00 B)

In [ ]:
embedding_docs[0]

array([   0,    0,    0,    0, 9118, 2198,  319, 5989], dtype=int32)

In [ ]:
model.predict(embedding_docs)

1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 126ms/step


array([[[ 0.02478928, -0.04533584, -0.01670442,  0.01774447,
         -0.01863657,  0.02845489,  0.04799161, -0.0377317 ,
          0.03857556,  0.03933707],
        [ 0.02478928, -0.04533584, -0.01670442,  0.01774447,
         -0.01863657,  0.02845489,  0.04799161, -0.0377317 ,
          0.03857556,  0.03933707],
        [ 0.02478928, -0.04533584, -0.01670442,  0.01774447,
         -0.01863657,  0.02845489,  0.04799161, -0.0377317 ,
          0.03857556,  0.03933707],
        [ 0.02478928, -0.04533584, -0.01670442,  0.01774447,
         -0.01863657,  0.02845489,  0.04799161, -0.0377317 ,
          0.03857556,  0.03933707],
        [-0.01309109,  0.04768822, -0.03044368, -0.04854254,
          0.01355766,  0.02145742, -0.0060522 ,  0.02181013,
          0.03410313, -0.03747837],
        [ 0.02005557, -0.04796566,  0.02414929,  0.0455379 ,
         -0.00514831, -0.0167082 ,  0.01169473,  0.02198369,
         -0.03054493,  0.04505967],
        [-0.04317423, -0.00290471, -0.04314464,  0.0